test/models/IllnessStartDateSchema.test.ts

```diff
@@ -0,0 +1,157 @@
+            new ValidationError(monthField, 'You must enter a month'),
+            new ValidationError(yearField, 'You must enter a year'),
+            new ValidationError(startDateField, 'Enter a real date')
+        ]);
+    });
+
+    it('should reject null values', () => {
+        const validationResult = validator.validate({
+            day: null,
+            month: null,
+            year: null,
+            illnessStart: null
+        });
+        assertValidationErrors(validationResult, [
+            new ValidationError(dayField, 'You must enter a day'),
+            new ValidationError(monthField, 'You must enter a month'),
+            new ValidationError(yearField, 'You must enter a year'),
+            new ValidationError(startDateField, 'Enter a real date')
+        ]);
+    });
+
+    it('should reject empty values', () => {
+        const validationResult = validator.validate({
+            day: '',
+            month: '',
+            year: '',
+            illnessStart: ''
+        });
+        assertValidationErrors(validationResult, [
+            new ValidationError(dayField, 'You must enter a day'),
+            new ValidationError(monthField, 'You must enter a month'),
+            new ValidationError(yearField, 'You must enter a year'),
+            new ValidationError(startDateField, 'Enter a real date')
+        ]);
+    });
+
+    it('should reject blank values', () => {
+        const validationResult = validator.validate({
+            day: ' ',
+            month: ' ',
+            year: ' ',
+            illnessStart: ' '
+        });
+        assertValidationErrors(validationResult, [
+            new ValidationError(dayField, 'You must enter a day'),
+            new ValidationError(monthField, 'You must enter a month'),
+            new ValidationError(yearField, 'You must enter a year'),
+            new ValidationError(startDateField, 'Enter a real date')
+        ]);
+    });
+
+    it('should reject invalid date patterns', () => {
+        const validationResult = validator.validate({
+            day: '001',
+            month: 'abc',
+            year: '*@&',
+            illnessStart: '0'
+        });
+        assertValidationErrors(validationResult, [
+            new ValidationError(dayField, 'You must enter a day'),
+            new ValidationError(monthField, 'You must enter a month'),
+            new ValidationError(yearField, 'You must enter a year'),
+            new ValidationError(startDateField, 'Enter a real date')
+        ]);
+    });
+
+
+    it('should reject invalid (non-existing) date', () => {
+        const validationResult = validator.validate({
+            day: '33',
+            month: '01',
+            year: '2020',
+            illnessStart: '2020-01-33'
+        });
+        assertValidationErrors(validationResult, [
+            new ValidationError(startDateField, 'Enter a real date')
+        ]);
+    });
+
+
+    it('should reject invalid (format) date', () => {
```

Comment:
What is wrong with that date? Seems legit to me.

Reply:
Alright, it is not about format, it is because date is in the past. Test name could be more descriptive.

Reply:
Is that because type is wrong? If so I got confused cause other examples that were invalid used string date as well.

Reply:
the type should be a valid ISO 8601 date format - i'll rename the test